scripts/check_compliance.py

```diff
@@ -165,6 +167,23 @@ def run(self):
         # single Kconfig tree and warns for all references to undefined symbols
         os.environ["KCONFIG_STRICT"] = "y"
 
+        kconfig_env_var_list_path = os.path.join(self.zephyr_base,
+                                    os.path.normpath("cmake/kconfig_env_vars.txt"))
+        # Skipping last new line which gives empty string
+        kconfig_env_var_list = open(kconfig_env_var_list_path).read().split("\n")[:-1]
+
+        # Faster to check for all than iterate over every variable
+        if not all(env_var in os.environ for env_var in kconfig_env_var_list):
+            missing_vars=[]
+            for env_var in kconfig_env_var_list:
+                if env_var not in os.environ.keys():
```

Comment:
Can do `if env_var not in os.environ:`. Iteration and membership testing always uses the keys.

Reply:
Changed to set instead